src/Infrastructure.VS/ThreadHandling.cs

```diff
@@ -42,6 +42,12 @@ public class ThreadHandling : IThreadHandling
 
         public T Run<T>(Func<Task<T>> asyncMethod) => ThreadHelper.JoinableTaskFactory.Run(asyncMethod);
 
+        public Task RunAsync<T>(Func<Task<T>> asyncMethod) => ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+        {
+            await Task.Yield(); // get off the caller's callstack.
```

Comment:
This is based on the cookbook although I'm not entirely sure we need it.
https://github.com/microsoft/vs-threading/blob/main/doc/cookbook_vs.md

Reply:
If we include the `Task.Yield()` call then we're not really wrapping the `JoinableTaskFactory` method, we're providing additional functionality. I think it would be better to provide either a simple wrapper that just does a pass-through, or come up with our own method with the functionality we want.
Given that we're not sure that we need the `Yield()` here, it might be better just to have the simple wrapper method.